### ISSUE TITLE
[BUG] When the feature to hide the taskbar is invoked, the size of the window is modified

### ISSUE THREAD

**Issue by Xiaokang2022** [Zingzy/hPyT#42]
**Describe the bug**

When the feature to hide the taskbar is invoked, the size of the window is modified. See the code below for details.

## Environment

```bash
$ python -m platform
Windows-11-10.0.26100-SP0

$ python --version
3.13.0

$ pip show hPyT
Name: hPyT
Version: 1.3.6
Summary: Hack Python Titlebar - A package to manipulate windows and titlebar of GUI applications made using python.
Home-page: https://github.com/zingzy/hPyT
Author: zingzy
Author-email: 
License: MIT
Location: C:\Software\Python313\Lib\site-packages
Requires: 
Required-by:
```

## Code

```python
import tkinter

import hPyT

root = tkinter.Tk()
root.geometry("200x200")

root.update_idletasks()
print(root.geometry())  # 200x200+52+52

hPyT.title_bar.hide(root)

root.update_idletasks()
print(root.geometry())  # 202x231+52+52

root.mainloop()
```

**Comment by Zingzy** [comment]
Hello @Xiaokang2022 thanks for pointing this Bug, I will take a look into this soon. Just a quick question, have you tried to re-create this BUG using some other GUI frameworks like customtkinter, PyQT, etc. ?

**Comment by Xiaokang2022** [comment]
customtkinter:

```python
import customtkinter
import hPyT

root = customtkinter.CTk()
root.geometry("200x200")

root.update_idletasks()
print(root.geometry())  # 133x133+152+152

hPyT.title_bar.hide(root)

root.update_idletasks()
print(root.geometry())  # 201x228+152+152
```

The results are similar.

**Comment by Zingzy** [comment]
Hello @Xiaokang2022 sorry I am not quite able to get what the BUG is, when we hide the titleBar the dimension of the content area is supposed to change as it spans the whole area which were used by window decorations lke the titlebar. Like for example

**Comment by Zingzy** [comment]
Hello @Xiaokang2022 sorry I am not quite able to get what the BUG is, when we hide the titleBar the dimension of the content area is supposed to change as it spans the whole area which were used by window decorations like the titlebar. For example:
```py
from customtkinter import CTk, CTkButton
from hPyT import title_bar

import ctypes
from ctypes import wintypes

user32: ctypes.WinDLL = ctypes.windll.user32

class RECT(ctypes.Structure):
    _fields_ = [
        ("left", wintypes.LONG),
        ("top", wintypes.LONG),
        ("right", wintypes.LONG),
        ("bottom", wintypes.LONG),
    ]


def get_window_rect(hwnd) -> RECT:  # returns the exact dimensions of the window including all of the decorations
    rect = RECT()
    user32.GetWindowRect(hwnd, ctypes.byref(rect))
    return rect

def show_window_dimensions():
    hwnd: int = ctypes.windll.user32.GetForegroundWindow()

    x_with_decorations: int = root.winfo_rootx()  # X position of the full window
    y_with_decorations: int = root.winfo_rooty()  # Y position of the full window

    x_without_decorations: int = root.winfo_x()  # X position of the content area
    y_without_decorations: int = root.winfo_y()  # Y position of the content area

    titlebar_height: int = y_with_decorations - y_without_decorations
    border_width: int = x_with_decorations - x_without_decorations

    window_rect: RECT = get_window_rect(hwnd)

    width: int = window_rect.right - window_rect.left
    height: int = window_rect.bottom - window_rect.top

    print(f"Title bar height: {titlebar_height}")
    print(f"Border width: {border_width}")
    print(f"Window including decorations - Width: {width}, Height: {height}")
    print(f"Content window dimensions: {root.winfo_geometry()}") # This will return the dimensions of the content area only

root = CTk()
root.title("Custom Tkinter")
root.wm_geometry("400x400")
root.update_idletasks()

def click(e=None):
    root.update_idletasks()

    print("------ Before hiding title bar ------")
    show_window_dimensions()

    title_bar.hide(root)

    print("------ After hiding title bar ------")
    show_window_dimensions()

button = CTkButton(root, text="Click Me", command=click)
button.place(relx=0.5, rely=0.5, anchor="center")

root.mainloop()
```

when the button is clicked this is what the output is:
```cmd
------ Before hiding title bar ------
Title bar height: 38
Border width: 9
Window including decorations - Width: 418, Height: 447
Content window dimensions: 400x400+128+128
------ After hiding title bar ------
Title bar height: 2
Border width: 8
Window including decorations - Width: 418, Height: 447
Content window dimensions: 402x437+128+128
```

Therefore, the original dimension of the window is not changed, it is the dimension of the content area that changes. Hope this clears some things up 😄

**Comment by Xiaokang2022** [comment]
But then I can't get the window size I want, unless I try it over and over again. Because the size of the window is different from the parameters I passed in.

In other words, I would expect the effect to be that the size of the content window remains the same when the title bar is removed.

**Comment by Xiaokang2022** [comment]
If this is not convenient to implement, then perhaps it should be pointed out in the documentation. Of course, it would be best to be able to meet this need.

**Comment by Zingzy** [comment]
> But then I can't get the window size I want, unless I try it over and over again. Because the size of the window is different from the parameters I passed in.
> 
> In other words, I would expect the effect to be that the size of the content window remains the same when the title bar is removed.

Maybe you can initially set the height of the window to be `height - titleBar height`? The code to get the TitleBar height:
```py
x_with_decorations: int = root.winfo_rootx()  # X position of the full window
y_with_decorations: int = root.winfo_rooty()  # Y position of the full window

x_without_decorations: int = root.winfo_x()  # X position of the content area
y_without_decorations: int = root.winfo_y()  # Y position of the content area

titlebar_height: int = y_with_decorations - y_without_decorations
```

**Comment by Xiaokang2022** [comment]
> We can also add a parameter in the `title_bar.hide` function which resizes the window to the expected height (window height - titlebar height)

That sounds good.

**Comment by Zingzy** [comment]
Alright then, I would mention about this ambiguity in the documentation and also implement this optional parameter in the next update!

**Comment by Zingzy** [comment]
Hello @Xiaokang2022 This Issue was solved in #44. The changes would be reflected in the next release `v1.3.7`. Meanwhile please check the [documentation](https://github.com/Zingzy/hPyT?tab=readme-ov-file#important-note-when-hiding-the-titlebar) I have added for this issue.
